### PR TITLE
Fix issue #9421: broken conversion within ORecordLazyList

### PR DESCRIPTION
**Pull request by PhantomYdn**
Issue #9421 is pretty annoying for our project. @tglman, @laa, @luigidellaquila, please, please, help with merging to `develop`.

### CODE REVIEW

**Comment by PhantomYdn** [comment]
Also let me mention that there was a difference between `public void add(int index, OIdentifiable e)` and `public boolean add(OIdentifiable e)`: last one did not invoke `ORecordInternal.track(sourceRecord, e);`. If I'm right - that was also a mistake. If I'm not right and in case `public boolean add(OIdentifiable e)` tracking should not be invoked (so it was OK) - please let me know.

**Comment by PhantomYdn** [comment]
Travis CI failed due to some other floating problem: it was observed on previous builds as well. For example: https://travis-ci.org/github/orientechnologies/orientdb/builds/739208716

Locally all tests are good.

File: core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java [hunk 1]
```diff
@@ -154,20 +154,23 @@ public boolean contains(final Object o) {
...
 
   @Override
   public void add(int index, OIdentifiable e) {
+    preAdd(e);
+    super.add(index, e);
+  }
+  
+  @Override
+  public boolean addInternal(OIdentifiable e) {
+	preAdd(e);
```
Comment (tglman) [inline core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java:171]: in this case the lazy load should not run, so is not correct to use preAdd as it is here.

**Review by tglman — CHANGES_REQUESTED** [review]
Looks correct, it just need to remove the lazy load logic in addInternal, and make sure the formatting is correct.

File: core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java (hunk 1 above)
Comment (PhantomYdn) [reply core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java:171]: @tglman , that's actually the actual fix. Without it - my testcase will fail.

Comment (PhantomYdn) [reply core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java:171]: Without calling `preAdd(a)` - list will not update `contentType` and first iterator will switch `contentType` from EMPTY to ALL_RECORDS which prevents the conversion of the rest of the items in the list.

Comment (PhantomYdn) [reply core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java:171]: Next commit is fixing this

**Comment by PhantomYdn** [comment]
@tglman , We really needs this in 3.1.5. Please consider merging before releasing the next version.

**Review by tglman — APPROVED** [review]

**Pull request merged**